azure_ad_admin/icon_azure_ad_admin/triggers/risk_detection/trigger.py

```diff
@@ -0,0 +1,78 @@
+import komand
+from .schema import RiskDetectionInput, RiskDetectionOutput, Input, Output, Component
+
+# Custom imports below
+from icon_azure_ad_admin.util.komand_clean_with_nulls import remove_null_and_clean
+from komand.exceptions import PluginException
+import time
+import requests
+import json
+
+
+class RiskDetection(komand.Trigger):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+                name='risk_detection',
+                description=Component.DESCRIPTION,
+                input=RiskDetectionInput(),
+                output=RiskDetectionOutput())
+        self.risk_level = ""
+        self.found = {}
+
+    def initialize(self):
+        new_risks = self.get_risks()
+
+        try:
+            result = new_risks['value']
+        except KeyError:
+            raise PluginException(cause='Unexpected output format.',
+                                  assistance='The output from Azure Active Directory was not in the expected format. Please contact support for help.',
+                                  data=result)
```

Comment:
result should not be used here. result is defined as new_risks['value'] in the try block. If an exception is raised it will be because new_risks['value] does not exist. Therefor result will be undefined. I assume it will default to NULL but I'm not sure. I would guess you should use new_risks as the value for data.

PS by unwritten convention our team defines the output variable from a requests call as result. So both luke and joey were instinctively reading this a requests object. This is why they both put result.text